pkg/smart_wallet/factory_calldata.go

```diff
@@ -0,0 +1,70 @@
+package smart_wallet
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/shopspring/decimal"
+)
+
+func GetFactoryCallData(smartWalletConfig Config, ownerAddress common.Address, index decimal.Decimal) ([]byte, error) {
+	var initCode []byte
+	var err error
+
+	switch typ := *smartWalletConfig.Type; typ {
```

Comment:
Now that the function is moved from `userop` package and is no longer private, we have no `nil` pointer "safety guards" and therefore it should have an associated default value. So, for example, if `smartWalletConfig.Type` is `nil`, the function interprets it as `KernelType`.